lib/utils.js

```diff
@@ -607,8 +607,8 @@ function decorateWithCollation(command, target, options) {
  * @param {object} command the command on which to apply the read concern
  * @param {Collection} coll the parent collection of the operation calling this method
  */
-function decorateWithReadConcern(command, coll) {
-  let readConcern = Object.assign({}, command.readConcern || {});
+function decorateWithReadConcern(command, coll, options) {
+  let readConcern = Object.assign({}, command.readConcern || (options ? options.readConcern : {}));
```

Comment:
I think you can just do `options ? options.readConcern : undefined` to reduce the extra needless allocation